src/ray/gcs/actor_state_accessor.cc

```diff
@@ -91,17 +91,37 @@ Status ActorStateAccessor::AsyncUpdate(const ActorID &actor_id,
 }
 
 Status ActorStateAccessor::AsyncSubscribe(
-    const SubscribeCallback<ActorID, ActorTableData> &subscribe,
+    const SubscribePairCallback<ActorID, ActorTableData> &subscribe,
     const StatusCallback &done) {
-  RAY_CHECK(subscribe != nullptr);
-  return actor_sub_executor_.AsyncSubscribe(ClientID::Nil(), subscribe, done);
+  return DoAsyncSubscribe(ActorID::Nil(), subscribe, done);
 }
 
 Status ActorStateAccessor::AsyncSubscribe(
-    const ActorID &actor_id, const SubscribeCallback<ActorID, ActorTableData> &subscribe,
+    const ActorID &actor_id,
+    const SubscribePairCallback<ActorID, ActorTableData> &subscribe,
+    const StatusCallback &done) {
+  return DoAsyncSubscribe(actor_id, subscribe, done);
+}
+
+Status ActorStateAccessor::DoAsyncSubscribe(
+    const ActorID &actor_id,
+    const SubscribePairCallback<ActorID, ActorTableData> &subscribe,
     const StatusCallback &done) {
   RAY_CHECK(subscribe != nullptr);
-  return actor_sub_executor_.AsyncSubscribe(node_id_, actor_id, subscribe, done);
+  auto on_subscribe = [subscribe](const ActorID &actor_id,
+                                  const std::vector<ActorTableData> &data) {
+    if (!data.empty()) {
+      // We only need the last entry, because it represents the latest state of
+      // this actor.
+      subscribe(actor_id, data.back());
+    }
+  };
+
+  if (actor_id.IsNil()) {
```

Comment:
Leave a comment to explain about this `if`.

Reply:
done.